Apparent race condition while updating the API Gateway account

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version

`0.11.13`

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* aws_api_gateway_account

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
# Role attached to API Gateway
resource "aws_iam_role" "api_gw_logging" {
  name = "api-gw-log-role"

  assume_role_policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": "sts:AssumeRole",
      "Principal": {
        "Service": "apigateway.amazonaws.com"
      },
      "Effect": "Allow",
      "Sid": ""
    }
  ]
}
EOF
}

# Attach AmazonAPIGatewayPushToCloudWatchLogs policy to API Gateway role to allow it to write logs
resource "aws_iam_role_policy_attachment" "attach_cloudwatch_logging" {
  role       = "${aws_iam_role.api_gw_logging.name}"
  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs"
}

# Enable logging for the API Gateway
resource "aws_api_gateway_account" "main" {
  cloudwatch_role_arn = "${aws_iam_role.api_gw_logging.arn}"
}
```

### Expected Behavior

The Terraform configuration should have been applied with no errors.

### Actual Behavior

Got error: 

```
1 error(s) occurred:
* aws_api_gateway_account.main: 1 error(s) occurred:
* aws_api_gateway_account.main: Updating API Gateway Account failed: BadRequestException: The role ARN does not have required permissions configured. Please grant trust permission for API Gateway and add the required role policy.
```

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

I have noticed that if I ran this command after Terraform failed, on the next run, Terraform would succeed:

```
# aws apigateway update-account --patch-operations op='replace',path='/cloudwatchRoleArn',value='arn:aws:iam::xxx:role/api-gw-log-role'
{
    "cloudwatchRoleArn": "arn:aws:iam::xxx:role/api-gw-log-role",
    "throttleSettings": {
        "burstLimit": 5000,
        "rateLimit": 10000.0
    },
    "features": [
        "UsagePlans"
    ],
    "apiKeyVersion": "4"
}
```

If I destroyed the Terraform deployment and ran that command right after, I would get the same error as Terraform:

```
# aws apigateway update-account --patch-operations op='replace',path='/cloudwatchRoleArn',value='arn:aws:iam::xxx:role/api-gw-log-role'
An error occurred (BadRequestException) when calling the UpdateAccount operation: The role ARN does not have required permissions configured. Please grant trust permission for API Gateway and add the required role policy.
```

So it looks like that error only came up when `api-gw-log-role` didn't exist, which made me think something was wrong with the role at the time it's being associated with `aws_api_gateway_account`. By looking at the Terraform build output, I could see that `aws_api_gateway_account` was being created before the `AmazonAPIGatewayPushToCloudWatchLogs` was attached:

```
aws_api_gateway_account.main: Creating...
  cloudwatch_role_arn: "" => "arn:aws:iam::788123154534:role/api-gw-log-role"
  throttle_settings.#: "" => "<computed>"
aws_iam_role_policy_attachment.attach_cloudwatch_logging: Creating...
  policy_arn: "" => "arn:aws:iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs"
  role:       "" => "api-gw-log-role"
```

For this reason, I have updated the `aws_api_gateway_account` configuration to explicitly depend on `aws_iam_role_policy_attachment.attach_cloudwatch_logging`:

```hcl
resource "aws_api_gateway_account" "main" {
  depends_on = ["aws_iam_role_policy_attachment.attach_cloudwatch_logging"] 
  cloudwatch_role_arn = "${aws_iam_role.api_gw_logging.arn}"
}
```

With that change, I consistently get the `BadRequestException` error on the first run, but a success on the next run. This is as far as I could get, so I'd appreciate if you could assist. 

I also want to clarify that I use the same configuration on various projects, but I'm only getting this issue consistently on a single project, which is what I normally use to test configuration changes. 

Thanks!

At first I thought this might be related to #11682 because of the timing, but we're noticing the same in our daily acceptance testing starting 4 days ago (which was a day after #11682 started and we'd be seeing empty error messages instead):

```
--- FAIL: TestAccAWSAPIGatewayAccount_basic (7.77s)
    testing.go:640: Step 0 error: errors during apply:
        
        Error: Updating API Gateway Account failed: BadRequestException: The role ARN does not have required permissions configured. Please grant trust permission for API Gateway and add the required role policy.
```

The `aws_api_gateway_account` resource does attempt to retry for IAM eventual consistency, however it looks like the error messaging changed slightly from:

```
The role ARN does not have required permissions set to API Gateway
```

to

```
The role ARN does not have required permissions configured
```

Would you be interested in patching this? Here's the line that can be adjusted (my recommendation would be to remove ` set to API Gateway` so its still compatible with the older AWS API error message:

https://github.com/terraform-providers/terraform-provider-aws/blob/96840caea28e3abf72d0d994550d2ee7d4b7fc45/aws/resource_aws_api_gateway_account.go#L96

--- a/aws/resource_aws_api_gateway_account.go
+++ b/aws/resource_aws_api_gateway_account.go
@@ -93,7 +93,7 @@ func resourceAwsApiGatewayAccountUpdate(d *schema.ResourceData, meta interface{}
 	log.Printf("[INFO] Updating API Gateway Account: %s", input)
 
 	// Retry due to eventual consistency of IAM
-	expectedErrMsg := "The role ARN does not have required permissions set to API Gateway"
+	expectedErrMsg := "The role ARN does not have required permissions"
 	otherErrMsg := "API Gateway could not successfully write to CloudWatch Logs using the ARN specified"
 	var out *apigateway.Account
 	var err error

